village/src/components/SmurfForm.js

```diff
@@ -1,28 +1,36 @@
 import React, { Component } from 'react';
+import Axios from 'axios';
 
+const theSmurfsURL = 'http://localhost:3333/smurfs';
 class SmurfForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
+      smurfs: [],
+      smurf:{
       name: '',
       age: '',
       height: ''
+      }
     };
   }
 
   addSmurf = event => {
     event.preventDefault();
     // add code to create the smurf using the api
-
-    this.setState({
-      name: '',
-      age: '',
-      height: ''
+   Axios.post(theSmurfsURL, this.state.smurf)
+        .then(res => {this.setState({ smurfs: [...res.data],  smurf: { 
```

Comment:
in this line you have used a nice syntax.
Could you explain your choice of the [...res.data] in this line

Reply:
that is a spread operator and I used it to take an existing array  from Smurf data  which can aid if I add another element, so it will still preserve the original array